components/alert/alert.js

```diff
@@ -0,0 +1,156 @@
+import '../button/button-icon.js';
+import '../button/button-subtle.js';
+import '../colors/colors.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { bodyCompactStyles } from '../typography/styles.js';
+import { LocalizeStaticMixin } from '../../mixins/localize-static-mixin.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+class Alert extends LocalizeStaticMixin(RtlMixin(LitElement)) {
+
+	static get properties() {
+		return {
+			buttonText: { type: String, attribute: 'button-text' },
+			hasCloseButton: { type: Boolean, attribute: 'has-close-button' },
+			hidden: { type: Boolean, reflect: true },
```

Comment:
Interesting... `hidden` is a standard DOM attribute that all nodes have. I don't _think_ we actually need to specify it? The only reason we might need to would be if reflection isn't working without it.

Reply:
A bit more on this: normally we like for people to manipulate attributes instead of call methods on components. i.e. `<d2l-dialog open>` instead of `open()`. So if `hidden` is the attribute/property equivalent of not needing to call the `close()` method, then I'm more OK with it. But I wish we'd called it something other than `hidden`, which is already a thing.

Reply:
In the Polymer version it used `setAttribute('hidden', 'hidden')` within `close`, I changed it to a reflected property so we didn't need to set the attribute ourselves.

As for the CSS style, I included it because it doesn't work in the demo pages so the close button looks broken. It does however work without the additional `css` in the LMS. I'll move the CSS so it only applies to the demo page rather than the component itself.

Reply:
The CSS is definitely needed if we're changing the default `display` to anything other than `inline` -- otherwise the browser's built-in CSS for the `display` attribute don't apply.